### PR TITLE
Delete DAG_FILE_NAME constant and use the resolve_flow_path function to obtain the flow path and flow file.

### PR DESCRIPTION
**Pull request by Stephen1993**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

Base this [PR](https://github.com/microsoft/promptflow/pull/2626): Distinguish flex flow and dag flow(use flow.flex.yaml for flex flow, flow.dag.yaml for dag flow),
And next [PR](https://github.com/microsoft/promptflow/pull/2548): Migrate the yaml used by the eager flow to flow.flex.yaml, code support both flow.flex.yaml and flow.dag.yaml. 

To standardize the construction method of flow paths and flow files:
1. Delete **DAG_FILE_NAME** constant and use the **resolve_flow_path** function to obtain the flow path and flow file.
2. Using **FLOW_DAG_YAML** instead of **DAG_FILE_NAME** .

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_proxy/_base_executor_proxy.py [hunk 1]
```diff
@@ -240,8 +239,9 @@ def get_inputs_definition(self):
         """Get the inputs definition of an eager flow"""
         from promptflow.contracts.flow import FlowInputDefinition
 
+        _, flow_file = resolve_flow_path(self.working_dir, check_flow_exist=False)
         flow_meta = self.generate_flow_json(
-            flow_file=self.working_dir / DAG_FILE_NAME,
+            flow_file=self.working_dir / flow_file,
```
Comment (Stephen1993) [inline src/promptflow-devkit/promptflow/_proxy/_base_executor_proxy.py:244]: Fixed to **FLOW_FLEX_YAML**?
Need the executor colleague help to confirm.

**Review by thy09 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_proxy/_base_executor_proxy.py (hunk 1 above)
Comment (Stephen1993) [reply src/promptflow-devkit/promptflow/_proxy/_base_executor_proxy.py:244]: Confirmed, only Flex flow uses this function.

**Review by wangchao1230 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Comment by daniellok-db** [comment]
cc @Stephen1993 @brynn-code could you also help update the [MLflow promptflow flavor](https://github.com/mlflow/mlflow/blob/c2d13b1bb9f84fb37bba57026026508c58f0326f/mlflow/promptflow/__init__.py#L278) to use this new API? The dev-version promptflow tests are currently failing after this change:

https://github.com/mlflow-automation/mlflow/actions/runs/8616373461/job/23624920227#step:12:1199

**Comment by brynn-code** [comment]
@daniellok-db Thanks for the reach out, we have add this constant back: #2731

**Pull request merged**